private/buf/bufmigrate/migrator.go

```diff
@@ -766,3 +775,17 @@ func filterFileSamePhpGenericServicesMap(m map[string][]string) map[string][]str
 	}
 	return c
 }
+
+// This is a helper function used to deduplicate ModuleKey and ModuleRef, both of
```

Comment:
If we wanted to do this, this could be accomplished with `slicesext.ToUniqueSorted(slicesext.Map(moduleKeys, bufmodule.ModuleKey.String))`. However, I have doubts that this is what we want.

Reply:
This would give us the slice of strings, we still need `ModuleRef`/`ModuleKey`.

Reply:
Got it - I just added `slicesext.Deduplicate` and `slicesext.DeduplicateAny`, you can do `slicesext.DeduplicateAny(moduleKeys, bufmodule.ModuleKey.String)` now